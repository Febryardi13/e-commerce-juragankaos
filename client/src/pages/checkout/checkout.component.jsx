import React from 'react'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'

import CheckoutItem from '../../components/checkout-item/checkout-item.component'
import StripeButton from '../../components/stripe-button/stripe-button.component'

import { selectCartItems, selectCartTotal } from '../../redux/cart/cart.selectors'

import { 
    CheckoutContainer,
    HeaderContainer,
    HeaderBlockContainer,
    TotalContainer,
    TestWarningContainer
} from './checkout.styles'

export const Checkout = ({ cartItems, total }) => (
    <CheckoutContainer>
        <HeaderContainer>
            <HeaderBlockContainer>
                <span>Product</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer>
                <span>Description</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer>
                <span>Quantity</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer>
                <span>Price</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer>
                <span>Remove</span>
            </HeaderBlockContainer>
        </HeaderContainer>
        {
            cartItems.map(cartItem => 
                (
                    <CheckoutItem key={cartItem.id} cartItem={cartItem} />
                )    
            )
        }
        <TotalContainer>
            <span>TOTAL: ${total}</span>
        </TotalContainer>
        <StripeButton price={total}/>
        <TestWarningContainer>
            *Please use the following test credit card for payments*
            <br/>
            <span> 4242 4242 4242 4242 </span>  
            <span>Exp: 07/20 -> current Date: MM/YY</span>
            <span> CVC: 123</span>
        </TestWarningContainer>
    </CheckoutContainer>
)

const mapStateToProps = createStructuredSelector ({
    cartItems: selectCartItems,
    total: selectCartTotal
})

export default connect(mapStateToProps)(Checkout)