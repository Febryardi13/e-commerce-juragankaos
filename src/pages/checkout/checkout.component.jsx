import React from 'react'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'

import CheckoutItem from '../../components/checkout-item/checkout-item.component'
import StripeButton from '../../components/stripe-button/stripe-button.component'

import { selectCartItems, selectCartTotal } from '../../redux/cart/cart.selectors'

import './checkout.styles.scss'

const Checkout = ({ cartItems, total }) => (
    <div className='checkout-page'>
        <div className='checkout-header'>
            <div className='heaader-block'>
                <span>Product</span>
            </div>
            <div className='heaader-block'>
                <span>Description</span>
            </div>
            <div className='heaader-block'>
                <span>Quantity</span>
            </div>
            <div className='heaader-block'>
                <span>Price</span>
            </div>
            <div className='heaader-block'>
                <span>Remove</span>
            </div>
        </div>
        {
            cartItems.map(cartItem => 
                (
                    <CheckoutItem key={cartItem.id} cartItem={cartItem} />
                )    
            )
        }
        <div className='total'>
            <span>TOTAL: ${total}</span>
        </div>
        <StripeButton price={total}/>
        <div className='test-warning'>
            *Please use the following test credit card for payments*
            <br/>
            <span> 4242 4242 4242 4242 </span>  
            <span>Exp: 07/20 -> current Date: MM/YY</span>
            <span> CVC: 123</span>
        </div>
    </div>
)

const mapStateToProps = createStructuredSelector ({
    cartItems: selectCartItems,
    total: selectCartTotal
})

export default connect(mapStateToProps)(Checkout)