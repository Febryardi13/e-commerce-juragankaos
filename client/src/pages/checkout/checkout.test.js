import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { Checkout } from './checkout.component'

configure({ adapter: new Adapter })

describe('Checkout component', () => {
    let wrapper;
    
    beforeEach(() => {
        const mockProps  ={
            cartItems: [],
            total: 100
        }
    
        wrapper = shallow(<Checkout {...mockProps} />)
    })
    
    it('should render Checkout component', () => {
        expect(wrapper).toMatchSnapshot()
    })
})

