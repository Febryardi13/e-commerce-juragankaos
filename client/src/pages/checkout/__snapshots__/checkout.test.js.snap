// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout component should render Checkout component 1`] = `ShallowWrapper {}`;
